condor_watch_q.py

```diff
@@ -420,6 +417,34 @@ def cli():
 GROUPBY_AD_KEY_TO_ATTRIBUTE = {v: k for k, v in GROUPBY_ATTRIBUTE_TO_AD_KEY.items()}
 
 
+def group_clusters(clusters, key, dagman_clusters_to_paths, batch_names, check_dagman):
+    groups = collections.defaultdict(list)
+    getter = operator.attrgetter(GROUPBY_AD_KEY_TO_ATTRIBUTE[key])
+    dagman_path_to_clusters = {v: k for k, v in dagman_clusters_to_paths.items()}
+
+    for cluster in clusters:
+        cluster_id = cluster.cluster_id
+        cluster_path = cluster.event_log_path
+
+        if cluster_path in dagman_path_to_clusters:
+            if key == BATCH_NAME:
+                dag_name = dagman_path_to_clusters[cluster_path]
+                groups[batch_names[dag_name]].append(cluster)
+            elif key == EVENT_LOG:
+                groups[cluster_path].append(cluster)
+            else:
+                groups[cluster_id].append(cluster)
+        else:
+            # Track only regular jobs, ignore dagman jobs
+            if cluster_id not in check_dagman:
```

Comment:
I think `check_dagman` should be called something like `dagman_job_cluster_ids`... that's a little wordy, but I want to make sure we capture what we're checking against.